app/views/handlers/question.py

```diff
@@ -195,12 +210,12 @@ def handle_post(self):
         # pylint: disable=no-member
         # wtforms Form parents are not discoverable in the 2.3.3 implementation
         self.questionnaire_store_updater.update_answers(self.form.data)
-        self.questionnaire_store_updater.update_progress_for_dependent_sections()
         if self.questionnaire_store_updater.is_dirty():
             self._routing_path = self.router.routing_path(
                 section_id=self._current_location.section_id,
                 list_item_id=self._current_location.list_item_id,
             )
+        self.questionnaire_store_updater.update_progress_for_dependent_sections()
```

Comment:
This would result in a duplicate call as the super handle post also calls this.

Reply:
It doesn't update progress for dependent sections in some scenarios when we rely on original `handle_post` `update_progress_for_dependent_sections` (some edit answers).  Dependant sections are not re-enabled if we don't run `update_progress_for_dependent_sections` before calling `super().handle_post()`.

Reply:
It seems like we don't do the capture step on edit answers as answers are not updated so this `if` doesn't apply:
```python
answer_updated = self._update_answer(
                answer_id_to_use, list_item_id_to_use, answer_value
            )
            if answer_updated:
                self._capture_section_dependencies_for_answer(answer_id_to_use)
                self._capture_block_dependencies_for_answer(answer_id_to_use)
```

Reply:
Hard to explain, it's just that if you don't add that line, which looks like duplication, it is not called at all. Probably best if we have a chat.

Reply:
This one is done now, fixed. It was order of method calls in block's handle_post that was incorrect. Needs to be:
```python
    def handle_post(self):
        self._set_started_at_metadata()
        self.questionnaire_store_updater.add_completed_location()
        self.questionnaire_store_updater.update_progress_for_dependent_sections()
        self._update_section_completeness()
        self.questionnaire_store_updater.save()
```
Instead of:
```python
    def handle_post(self):
        self._set_started_at_metadata()
        self.questionnaire_store_updater.add_completed_location()
        self._update_section_completeness()
        self.questionnaire_store_updater.update_progress_for_dependent_sections()
        self.questionnaire_store_updater.save()
```

Reply:
This would break it for @berroar's progress to change. As dependent section will be using the old status.

Reply:
Addressed.